xray/commands/curation/audit.go

```diff
@@ -0,0 +1,513 @@
+	}()
+	consumerProducer.Run()
+
+	if err := errorsQueue.GetError(); err != nil {
+		multiErrors = errors.Join(err, multiErrors)
+	}
+	return multiErrors
+}
+
+func (nc *treeAnalyzer) fetchNodeStatus(node xrayUtils.GraphNode, p *sync.Map /*drainChan chan *PackageStatus*/) error {
+	packageUrl, name, version := getUrlNameAndVersionByTech(nc.tech, node.Id, nc.url, nc.repo)
+	resp, _, err := nc.artiManager.Client().SendHead(packageUrl, &nc.httpClientDetails)
+	if err != nil {
+		if resp == nil || resp.StatusCode != http.StatusForbidden {
+			return err
+		}
+	}
+	if resp != nil && resp.StatusCode >= 400 && resp.StatusCode != http.StatusForbidden {
+		return fmt.Errorf("Failed sending HEAD to %s for package %s. Status-code: %v", packageUrl, node.Id, resp.StatusCode)
+	}
+	if resp.StatusCode == http.StatusForbidden {
+		pkStatus, err := nc.getBlockedPackageDetails(packageUrl, name, version)
+		if err != nil {
+			return err
+		}
+		if pkStatus != nil {
+			p.Store(pkStatus.BlockedPackageUrl, pkStatus)
+		}
+	}
+	return nil
+}
+
+// We try to collect curation details from GET response after HEAD request got forbidden status code.
+func (nc *treeAnalyzer) getBlockedPackageDetails(packageUrl string, name string, version string) (*PackageStatus, error) {
+	getResp, respBody, _, err := nc.artiManager.Client().SendGet(packageUrl, true, &nc.httpClientDetails)
+	if err != nil {
+		if getResp == nil {
+			return nil, err
+		}
+		if getResp.StatusCode != http.StatusForbidden {
+			return nil, fmt.Errorf("Failed sending HEAD request to %s for package '%s:%s'. "+
+				"Status code: %v. Cause: %v", packageUrl, name, version, getResp.StatusCode, err)
+		}
+	}
+	if getResp.StatusCode == http.StatusForbidden {
+		respError := &ErrorsResp{}
+		if err := json.Unmarshal(respBody, respError); err != nil {
+			log.Error(fmt.Sprintf("failed to decode response for 'forbidden' response, err: %v", err))
+			return nil, nil
+		}
+		if len(respError.Errors) == 0 {
+			return nil, fmt.Errorf("No errors messages for package %s, version: %s for download url: %s ", name, version, packageUrl)
+		}
+		// if the error message contains the curation string key, then we can be sure it got blocked by curation service.
+		if strings.Contains(strings.ToLower(respError.Errors[0].Message), BlockMessageKey) {
+			blockingReason := BlockingReasonPolicy
+			if strings.Contains(strings.ToLower(respError.Errors[0].Message), NotBeingFoundKey) {
+				blockingReason = BlockingReasonNotFound
+			}
+			policies := extractPoliciesFromMsg(respError)
+			return &PackageStatus{
+				PackageName:       name,
+				PackageVersion:    version,
+				BlockedPackageUrl: packageUrl,
+				Action:            blocked,
+				Policy:            policies,
+				BlockingReason:    blockingReason,
+				PkgType:           string(nc.tech),
+			}, nil
+		}
+	}
+	return nil, nil
+}
+
+// Return policies and conditions names from forbidden error message. The error message structure happens when package was blocked by curation.
```

Comment:
...from forbidden error message... --> ...from the FORBIDDEN HTTP error message. ...

Let's also remove this line, since it's wrong in terms of grammar, and doesn't add much information - 
**The error message structure happens when package was blocked by curation.**